batch7crud-roll5/src/main/java/com/lftechnology/batch7crud/controller/EmployeeController.java

```diff
@@ -0,0 +1,199 @@
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet implementation class EmployeeController
+ */
+
+@WebServlet("/employees/*")
+public class EmployeeController extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+    
+    EmployeeService employeeService = new EmployeeService();
+    Employee emp = new Employee();
+    public static String employeeListUrl = "/batch7crud-roll5/employees";
+
+    /**
+     * @see HttpServlet#HttpServlet()
+     */
+    public EmployeeController() {
+        super();
+        // TODO Auto-generated constructor stub
+    }
+
+    /**
+     * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
+     */
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        // TODO Auto-generated method stub
+        String path = request.getPathInfo();
+        if (path != null) {
+
+            String[] pathParts = path.split("/");
+            
+            if ("create".equals(pathParts[1]))
+                create(request, response);
+
+            else if ("edit".equals(pathParts[2])) {
+                int id = Integer.parseInt(pathParts[1]);
+                edit(request, response, id);
+            }
+
+            else if ("page".equals(pathParts[1])) {
+                int pageNo = Integer.parseInt(pathParts[2]);
+                fetchData(request, response, pageNo - 1);
+            }
+
+        }
+
+        else {
+            int pageNo = 1;
+            fetchData(request, response, pageNo - 1);
+        }
+
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String path = request.getPathInfo();
+        if (path != null) {
+            String[] pathParts = path.split("/");
+
+            if ("createProcess".equals(pathParts[1])) {
+                createProcess(request, response);
+
+            }
+
+            else if ("editProcess".equals(pathParts[2])) {
+                int id = Integer.parseInt(pathParts[1]);
+                editProcess(request, response, id);
+
+            }
+
+            else if ("deleteProcess".equals(pathParts[2])) {
+                int id = Integer.parseInt(pathParts[1]);
+                deleteProcess(request, response, id);
+            }
+
+        } else {
+            int pageNo = 1;
+            fetchData(request, response, pageNo - 1);
+        }
+
+    }
+
+    private void deleteProcess(HttpServletRequest request, HttpServletResponse response, int id) {
+        // TODO Auto-generated method stub
+        try {
+            employeeService.deleteById(id);
+
+            response.sendRedirect(employeeListUrl);
+
+        } catch (Exception e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)